design/secrets.md

```diff
@@ -0,0 +1,81 @@
+# Design proposal template (replace with your proposal's title)
+
+Currently, Velero only supports a single AIM access credential secret per location provider/plugin. Velero creates and stores the plugin credential secret under the hard-coded key `secret.cloud-credentials.data.cloud`.
+
+This makes it so switching from one plugin to another necessitates overriding the existing credential secret with the appropriate one for the new plugin.
+
+## Goals
+
+- To allow Velero to create and store multiple secrets for provider credentials, even multiple credentials for the same provider
+- To improve the UX for configuring the velero deployment with multiple plugins/providers, and corresponding IAM secrets.
+
+## Non Goals
+
+- To make any change except what's necessary to handle multiple credentials
+
+## Design overview
+
+- Instead of one credential per Velero deployment, multiple credentials can be added and will be associated with plugins.
+  Credentials can be added when plugins are added or can be set later.
+  
+- When new credentials are added, the `cloud-credentials` secret will be updated with a new entry containing the new credentials value.
+  We will add keys to the existing secret, rather than creating new ones, due to the fact that it is not possible to modify volume mounts in a running pod or container.
+  Changes to secrets (such as adding new data keys) are automatically updated within the volume mount.
+  This means that any update to the `cloud-credentials` secret would automatically be available to the pods where that secret is mounted.
+  This allows us to work around the restriction of modifying volume mounts to instead still use a single secret, but add more keys to it for new sets of credentials.
+ 
+- BSLs and VSLs will be updated to contain a reference to a particular credential name, which will be a key in the `cloud-credentials` secret.
+  When a new BSL or VSL is created, a specific set of credentials can be selected for use.
+  The set of credentials associated with a BSL or VSL will be used when running the plugin processes needed for that storage location.
+
+## Detailed Design
+
+- The name of the flag changes from `secret-file` to `--credentials-file`.
+
+- The arguments to `velero plugin (add|set) --credentials-file` will be a map of the credentials name as a key, and the path to the file as a value.
+  This way, we can have multiple credential secrets and each secret per provider/plugin will be unique.
+  There are two ways in which we can choose to name the keys stored in the `cloud-credentials` secret:
+    1. Combine the name of the provider (extracted from the plugin name) and the given credentials name.
```

Comment:
If we do go with generated names, I would treat them all the same - the velero.io plugins should use the same rules as ones we don't ship, in my opinion, as we're going to remove the exception case eventually. That also means less code to worry about.